### DISCUSSION TITLE
How to detect internal package change and build app that depends on it, so changes are reflected?

### DISCUSSION THREAD

**Discussion by vittis** [vercel/turborepo#7441] [Help] [answered]
### Summary

Mostly followed these links:
https://turbo.build/repo/docs/handbook/sharing-code/internal-packages
https://turbo.build/repo/docs/handbook/publishing-packages/bundling

I have `apps/server` and `packages/logic`.

My `server` scripts looks like this:
```
"scripts": {
  "start": "node dist/index.js",
  "build": "tsup",
  "dev": "tsup --watch --onSuccess \"node dist/index.js\""
},
```

`logic` package.json (relevant parts):
```
{
	"main": "./dist/index.js",
	"module": "./dist/index.mjs",
	"types": "./dist/index.d.ts",
	"files": [
		"dist/**"
	],
	"scripts": {
		"build": "tsup game/index.ts --format cjs,esm --dts",
		"dev": "tsup game/index.ts --format cjs,esm --dts --watch",
       }
}
```

All of this works. But one very annoying thing that happens is that when I save a file on the `logic` package, the server app does not trigger a build, I have to manually save a file in `server` so the changes of `logic` reflect there.

Am I missing something? Is this expected?

### Example

https://github.com/vittis/shiny-potate

**Comment by anthonyshew** [comment, marked as answer]
Sounds like you're looking for this issue: https://github.com/vercel/turbo/issues/986

We're actively working on this so will close out here.
